modules/mobile/app/controllers/mobile/v0/pre_need_burial_controller.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'chip/service'
+require 'preneeds/service'
+
+module Mobile
+  module V0
+    class PreNeedBurialController < ApplicationController
+      FORM = '40-10007'
+
+      def create
+        validate!
+
+        @form = ::Preneeds::BurialForm.new(burial_form_params)
+        @resource = client.receive_pre_need_application(@form)
+
+        create_local_preneed_submission(@resource)
+
+        render json: Mobile::V0::PreNeedBurialSerializer.new(@resource)
+      end
+
+      private
+
+      def create_local_preneed_submission(resource)
+        ::Preneeds::PreneedSubmission.create!(
+          tracking_number: resource.tracking_number,
+          application_uuid: resource.application_uuid,
+          return_description: resource.return_description,
+          return_code: resource.return_code
+        )
+      end
+
+      def burial_form_params
+        params.require(:application).permit(
+          :application_status, :has_currently_buried, :sending_code,
+          preneed_attachments: ::Preneeds::PreneedAttachmentHash.permitted_params,
+          applicant: ::Preneeds::Applicant.permitted_params,
+          claimant: ::Preneeds::Claimant.permitted_params,
+          currently_buried_persons: ::Preneeds::CurrentlyBuriedPerson.permitted_params,
+          veteran: ::Preneeds::Veteran.permitted_params
+        )
+      end
+
+      def validate!
+        json_params = JSON.parse(params.to_json)
+        race_hash = json_params.dig('application', 'veteran', 'race')
+        json_params.dig('application', 'veteran', 'race').keys.map do |a|
+          race_hash[a] = ActiveRecord::Type::Boolean.new.cast(race_hash[a])
```

Comment:
one difference between our middleware is that it was coverting all hash values to strings, causing the booleans to through errors in the validation, this manually sets them back to booleans

Reply:
I think you can bypass the race hash and just do this:
```
json_params.dig('application', 'veteran', 'race').transform_values!  do |v|
  ActiveRecord::Type::Boolean.new.cast(v)
end
```